core/trino-main/src/main/java/io/trino/cost/AggregationStatsRule.java

```diff
@@ -54,10 +52,6 @@ protected Optional<PlanNodeStatsEstimate> doCalculate(AggregationNode node, Stat
             return Optional.empty();
         }
 
-        if (node.getStep() != SINGLE && node.getStep() != FINAL) {
```

Comment:
This will have significantly different estimates for final aggregation with/without PA disabled, right?
Final aggregation estimates should be the same as if PA is disabled.

Reply:
For query `explain analyze (select distinct extendedprice from lineitem);` Partitioned by `shipdate`: 
```
set session prefer_partial_aggregation=true;
Aggregate[type = FINAL, keys = [extendedprice], hash = [$hashvalue]]
Estimates: {rows: 25198 (442.93kB), cpu: 442.93k, memory: 442.93kB, network: 0B}
```
  vs
```
set session prefer_partial_aggregation=false;
Aggregate[keys = [extendedprice], hash = [$hashvalue]]
Estimates: {rows: 25198 (442.93kB), cpu: 995.72M, memory: 442.93kB, network: 0B}
```
For query `explain analyze (select count(*) from lineitem_t where extendedprice = (select max(extendedprice) from lineitem))`:
```
set session prefer_partial_aggregation=true;
Aggregate[type = FINAL]
Estimates: {rows: 1 (9B), cpu: 9, memory: 9B, network: 0B}                                                                                                  
```
vs
```
set session prefer_partial_aggregation=false;
Aggregate[]
Estimates: {rows: 1 (9B), cpu: 0, memory: 9B, network: 0B}
```

Reply:
Does it work for every case? Why? Add a comment explaining it if it's the case

Reply:
Q: Will it work for partitial aggregation that pushed through join? Also, won't it wrongly estimate the exchange stats which is right above partitial aggregation?